package AllServerLords;

message Lord {
	required uint32 uid			= 1;
	required uint32 type		= 2;
	required uint32 worship 	= 3;	
	required uint32 setted		= 4;
	required string	heroid		= 5;
	required string name 		= 6;
	required string headurl		= 7;
	required string heroname	= 8;
}

message LordItems {
	required uint32		id 		= 1;
	required uint32		time	= 2;
	required uint32		finished = 3;
	repeated Lord	lord 	= 4;
}

message LordRankItem {
	required uint32 uid			= 1;
	required string name 		= 2;
	required uint64	sumdata		= 3;
}

message LordRankList {
	required uint32 		type 	= 1;
	required uint32			time	= 2;
	repeated LordRankItem 	list	= 3;
}

message LordContainer {
	required uint32			curid	= 1;
	repeated LordItems  	lords = 2;
	repeated LordRankList   ranks = 3;
}

message LordAllZone {
	repeated LordContainer zones = 1;
}
